File: frontend/src/types/evaluation.ts
export interface ClassMetrics {
  label: string
  precision: number
  recall: number
  f1: number
  support: number
}

export interface EvaluationReport {
  name: string
  accuracy: number
  macro_precision: number
  macro_recall: number
  macro_f1: number
  weighted_f1: number
  cohen_kappa: number
  mae: number
  rmse: number
  per_class: ClassMetrics[]
  confusion_matrix: number[][]
  n_samples: number
}

export interface BenchmarkResponse {
  reports: EvaluationReport[]
  n_samples: number
  seed: number
}

export interface AblationResult {
  active_modalities: string[]
  dropped_modalities: string[]
  accuracy: number
  macro_f1: number
  weighted_f1: number
  cohen_kappa: number
}

export interface AblationStudyResponse {
  n_samples: number
  seed: number
  results: AblationResult[]
  baseline_f1: number
}

File: backend/app/schemas/evaluation.py
"""Pydantic schemas for evaluation API responses."""
from __future__ import annotations

from typing import Dict, List, Optional
from pydantic import BaseModel


class ClassMetricsSchema(BaseModel):
    label: str
    precision: float
    recall: float
    f1: float
    support: int


class EvaluationReportSchema(BaseModel):
    name: str
    accuracy: float
    macro_precision: float
    macro_recall: float
    macro_f1: float
    weighted_f1: float
    cohen_kappa: float
    mae: float
    rmse: float
    per_class: List[ClassMetricsSchema]
    confusion_matrix: List[List[int]]
    n_samples: int


class BenchmarkResponse(BaseModel):
    reports: List[EvaluationReportSchema]
    n_samples: int
    seed: int


class AblationResultSchema(BaseModel):
    active_modalities: List[str]
    dropped_modalities: List[str]
    accuracy: float
    macro_f1: float
    weighted_f1: float
    cohen_kappa: float


class AblationStudyResponse(BaseModel):
    n_samples: int
    seed: int
    results: List[AblationResultSchema]
    baseline_f1: float          # full-fusion F1 for comparison
